### PR TITLE
Add setPackageName and setAttributionTag to the Oboe builder

### LINKED ISSUE #1282: Add support for setPackageName(n)
There are new functions in AAudio:

- AAudioStreamBuilder_setPackageName()
- AAudioStreamBuilder_setAttributeTag()

Add Oboe access.
Also maybe set default AttributionTag to "Oboe" for Media Metrics.

### PR DESCRIPTION
**Pull request by robertwu1**
In Android 12, we added an ability to set Package Names and Attribution Tags to AAudio. We are also doing this in Oboe.
Note that getting the values comes from the context (Context#getPackageName()/Context#getAttributionTag())

Fixes #1282

Tested via:
Built OboeTester
Built and ran unit tests

### CODE REVIEW

**Comment by robertwu1** [comment]
It seems like certain types of streams don't play well with setting the package name. For example, Tap To Tone Latency works perfectly but Glitch Test has some weird bugs. When running the unit tests, some randomly fail setting the package name before 
 commit 4. @philburk should I assign a bug to Phillip Moltmann?

File: src/aaudio/AudioStreamAAudio.cpp [hunk 1]
```diff
@@ -300,6 +309,14 @@ Result AudioStreamAAudio::open() {
         mSessionId = SessionId::None;
     }
 
+    // These were added in S so we have to check for the function pointer.
+    //if (mLibLoader->stream_getPackageName != nullptr) {
```
Comment (philburk) [inline src/aaudio/AudioStreamAAudio.cpp:313]: Remove dead code.

**Review by philburk — CHANGES_REQUESTED** [review]

File: include/oboe/AudioStreamBase.h [hunk 2]
```diff
@@ -156,6 +157,16 @@ class AudioStreamBase {
      */
     SessionId getSessionId() const { return mSessionId; }
 
+    /**
+     * @return the package name for this stream.
+     */
+    std::string getPackageName() const { return mPackageName; }
```
[the remark this answers is not in the document]
Comment (robertwu1) [reply include/oboe/AudioStreamBase.h]: Thanks Phil!

**Comment by robertwu1** [comment]
@philburk Thanks for adding comments! I removed all of the setters. Let me know if this looks good

**Comment by dturner** [comment]
These features were added to AAudio to allow more detailed logging.

**Review by philburk — APPROVED** [review]

**Pull request merged**